### DISCUSSION TITLE
is there any current support for hw-aware NAS techniques?

### DISCUSSION THREAD

**Discussion by kaoutar55** [microsoft/nni#3761] [Q&A] [answered]
**What would you like to be added**:

**Why is this needed**:

**Without this feature, how does current nni work**：

**Components that may involve changes**:

**Brief description of your proposal if any**:

**Comment by QuanluZhang** [comment, marked as answer]
@kaoutar55 hardware-aware NAS is an interesting research topic. There are different ways to support hardware-aware NAS.

1. Adding hardware related metric into performance value of each sampled model. For example, adding inference latency on a given hardware to the model's accuracy, and report the combined performance value (using nni.report_final_result) to tuning algorithm. This is the simplest way and it is natively supported by NNI.

2. The search space is specifically tailored for a hardware (for example, using hardware friendly operators only). These are mainly handled by new hardware-aware NAS algorithms. NNI will cover this kind of algorithms in future release. We also encourage users to contribute such algorithms to NNI.
